internal/nfs/controller/volume.go

```diff
@@ -135,24 +136,33 @@ func (nv *NFSVolume) CreateExport(backend *csi.Volume) error {
 		return fmt.Errorf("failed to set NFS-cluster: %w", err)
 	}
 
-	// TODO: use new go-ceph API, see ceph/ceph-csi#2977
-	// new versions of Ceph use a different command, and the go-ceph API
-	// also seems to be different :-/
-	//
-	// run the new command, but fall back to the previous one in case of an
-	// error
-	cmds := [][]string{
-		// ceph nfs export create cephfs --cluster-id <cluster_id>
-		//     --pseudo-path <pseudo_path> --fsname <fsname>
-		//     [--readonly] [--path=/path/in/cephfs]
-		nv.createExportCommand("--cluster-id="+nfsCluster,
-			"--fsname="+fs, "--pseudo-path="+nv.GetExportPath(),
-			"--path="+path),
-		// ceph nfs export create cephfs ${FS} ${NFS} /${EXPORT} ${SUBVOL_PATH}
-		nv.createExportCommand(nfsCluster, fs, nv.GetExportPath(), path),
+	nfsa, err := nv.conn.GetNFSAdmin()
+	if err != nil {
+		return fmt.Errorf("failed to get NFSAdmin: %w", err)
+	}
+
+	_, err = nfsa.CreateCephFSExport(nfs.CephFSExportSpec{
+		FileSystemName: fs,
+		ClusterID:      nfsCluster,
+		PseudoPath:     nv.GetExportPath(),
+		Path:           path,
+	})
```

Comment:
As an enhancement may be we can store some bool value for what command is supported instead of calling invalid command for each new request.

Reply:
same for Remove also

Reply:
I don't think that will work. Hopefully users update their Ceph cluster at one point, the CLI will then fail, and the API wil work. We need to detect it for each command that is executed.

Reply:
it should be possible we used it for the cephfs resize operation.

Reply:
unfortunately, the `ceph nfs ...` commands have changed when the API was introduced. They did not keep backward compatibility.